Compiler/src/Error.hs

```diff
@@ -1,6 +1,19 @@
 module Error where
 
--- TODO: Add more specific error types
-data Error = ErrorMessage String
+import AST
+
+data Error = SyntaxError String
```

Comment:
I switched it so that `Error` is a typeclass and all of `LookupError`, `FlowError`, etc. are individual types which implement PrettyPrint. I think this should be a more flexible system as this way you can define multiple data constructors per error type.

Reply:
I think that it is still better to have it as a single `Error` type, because that allows us to test with the `Error` values, rather than the pretty-printed strings, which have the same issue as before. Unfortunately, Haskell doesn't have subtyping to give us the best of both worlds; what you can do is make a new type like:
```
data LookupErrorCat = LookupErrorVar | LookupErrorType | LookupErrorTypeDecl 
    deriving (Show, Eq)
```
and make this a parameter of the `LookupError` constructor. Also, we should put the `deriving (...)` on the next line, for consistency.